testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -130,29 +130,44 @@ impl SignerTest<SpawnedSigner> {
 
         info!("Got {} signatures", signature.len());
 
-        assert_eq!(signature.len(), num_signers);
+        // NOTE: signature.len() does not need to equal signers.len(); the stacks miner can finish the block
+        //  whenever it has crossed the threshold.
+        assert!(signature.len() >= num_signers * 7 / 10);
 
         let reward_cycle = self.get_current_reward_cycle();
         let signers = self.get_reward_set_signers(reward_cycle);
 
         // Verify that the signers signed the proposed block
-        let all_signed = signers.iter().zip(signature).all(|(signer, signature)| {
+        let mut signer_index = 0;
+        let mut signature_index = 0;
+        let validated = loop {
+            let Some(signature) = signature.get(signature_index) else {
+                break true;
```

Comment:
It took me a couple minutes to understand why `break true` makes sense here. Could you add a comment like this:

```rust
// Since we've already checked `signature.len()`, this means we've validated an acceptable number of signatures in this loop
```